tests/test_sanity.py

```diff
@@ -144,14 +149,60 @@ def test_from_container(
     runtime_specific_args: List[str],
     gprofiler_docker_image: Image,
     output_directory: Path,
+    output_collapsed: Path,
     assert_collapsed: AssertInCollapsed,
     assert_app_id: Callable,
     profiler_flags: List[str],
 ) -> None:
     _ = application_pid  # Fixture only used for running the application.
     _ = assert_app_id  # Required for mypy unused argument warning
     collapsed_text = run_gprofiler_in_container_for_one_session(
-        docker_client, gprofiler_docker_image, output_directory, runtime_specific_args, profiler_flags
+        docker_client, gprofiler_docker_image, output_directory, output_collapsed, runtime_specific_args, profiler_flags
     )
     collapsed = parse_one_collapsed(collapsed_text)
     assert_collapsed(collapsed)
+
+
+@pytest.mark.parametrize(
+    "runtime,profiler_type",
+    [
+        ("java", "ap"),
+        ("python", "py-spy"),
+        ("ruby", "rbspy"),
+    ],
+)
+def test_from_container_spawned_process(
+    docker_client: DockerClient,
+    runtime_specific_args: List[str],
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    assert_collapsed: AssertInCollapsed,
+    profiler_flags: List[str],
+    application_factory: Callable[[], _GeneratorContextManager],
+) -> None:
+    args = runtime_specific_args.copy()
+    if "-d" in args:
+        args[args.index("-d") + 1] = "10"
+    else:
+        args.extend(["-d", "10"])
+    profiler_flags.append("--profile-spawned-processes")
+    container = start_gprofiler_in_container_for_one_session(
+        docker_client,
+        gprofiler_docker_image,
+        output_directory,
+        output_collapsed,
+        args,
+        profiler_flags,
+    )
+
+    try:
+        wait_event(30, Event(), lambda: b"starting profiling spawning processes" in container.logs())
+    except TimeoutError:
+        print(container.logs())
+        raise
+
+    with application_factory():
+        collapsed_text = wait_for_gprofiler_container(container, output_collapsed)
```

Comment:
What proves that it's the "spawning processes" mechanism that caught the process is - we waited for gprofiler to do the first `_select_processes_to_profile()`  and only then we spawned the app, and gprofiler runs non-continuous so if app results exist in the final collapsed, we know it's due to spawning. Right? Maybe comment